paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PaymentOptionsUI.kt

```diff
@@ -0,0 +1,269 @@
+import com.stripe.android.paymentsheet.R
+import com.stripe.android.paymentsheet.model.PaymentSelection
+import com.stripe.android.paymentsheet.toPaymentSelection
+import com.stripe.android.paymentsheet.viewmodels.BaseSheetViewModel
+import com.stripe.android.uicore.shouldUseDarkDynamicColor
+import com.stripe.android.uicore.stripeColors
+
+@Composable
+internal fun PaymentOptions(
+    viewModel: BaseSheetViewModel,
+    modifier: Modifier = Modifier,
+) {
+    val context = LocalContext.current
+
+    val state by viewModel.paymentOptionsState.collectAsState()
+    val isEditing by viewModel.editing.collectAsState()
+    val isProcessing by viewModel.processing.collectAsState()
+
+    val nameProvider: (String?) -> String = remember {
+        { code ->
+            val repo = viewModel.lpmResourceRepository.getRepository()
+            val resource = repo.fromCode(code)?.displayNameResource
+            context.getString(resource!!)
+        }
+    }
+
+    PaymentOptions(
+        state = state,
+        nameProvider = nameProvider,
+        isEditing = isEditing,
+        isProcessing = isProcessing,
+        onAddCardPressed = viewModel::transitionToAddPaymentScreen,
+        onItemSelected = viewModel::handlePaymentMethodSelected,
+        onItemRemoved = viewModel::removePaymentMethod,
+        modifier = modifier,
+    )
+}
+
+@Composable
+internal fun PaymentOptions(
+    state: PaymentOptionsState,
+    nameProvider: (PaymentMethodCode?) -> String?,
+    isEditing: Boolean,
+    isProcessing: Boolean,
+    onAddCardPressed: () -> Unit,
+    onItemSelected: (PaymentSelection?) -> Unit,
+    onItemRemoved: (PaymentMethod) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    BoxWithConstraints(modifier = modifier.fillMaxWidth()) {
+        val width = rememberItemWidth(maxWidth)
+
+        LazyRow(
+            userScrollEnabled = !isProcessing,
+            contentPadding = PaddingValues(horizontal = 17.dp),
+        ) {
+            items(state.items) { item ->
+                val isEnabled = !isProcessing && (!isEditing || item.isEnabledDuringEditing)
+                val isSelected = item == state.selectedItem && !isEditing
+
+                PaymentOption(
+                    item = item,
+                    nameProvider = nameProvider,
+                    width = width,
+                    isEditing = isEditing,
+                    isEnabled = isEnabled,
+                    isSelected = isSelected,
+                    onAddCardPressed = onAddCardPressed,
+                    onItemSelected = onItemSelected,
+                    onItemRemoved = onItemRemoved,
+                    modifier = Modifier.testTag(item.viewType.name)
+                )
+            }
+        }
+    }
+}
+
+@Composable
+internal fun rememberItemWidth(maxWidth: Dp): Dp {
+    val targetWidth = maxWidth - 17.dp * 2
+    val minItemWidth = 100.dp + (6.dp * 2)
+    // numVisibleItems is incremented in steps of 0.5 items (1, 1.5, 2, 2.5, 3, ...)
```

Comment:
Could help to give some reasoning as to why here.

Reply:
I copied this over from `PaymentOptionsAdapter.calculateWidth()` and can’t really tell you why this works 😔